aspnetcore/client-side/import-export-interop.md

```diff
@@ -0,0 +1,394 @@
+
+```dotnetcli
+dotnet workload install wasm-tools
+```
+
+Optionally, install the `wasm-experimental` workload, which contains experimental project templates for getting started with .NET on WebAssembly in a browser app (WebAssembly Browser App) or in a Node.js-based console app (WebAssembly Console App). This workload isn't required if you plan to integrate `[JSImport]`/`[JSExport]` interop into an existing JS app.
+
+```dotnetcli
+dotnet workload install wasm-experimental
+```
+
+For more information, see the [Experimental workload and project templates](#experimental-workload-and-project-templates) section.
+
+## App configuration
+
+To configure the app's project file (`.csproj`):
+
+* Target `net7.0` or later:
+
+  ```xml
+  <TargetFramework>net7.0</TargetFramework>
+  ```
+
+* Specify `browser-wasm` for the runtime identifier:
+
+  ```xml
+  <RuntimeIdentifier>browser-wasm</RuntimeIdentifier>
+  ```
+
+* Specify an executable output type:
+
+  ```xml
+  <OutputType>Exe</OutputType>
+  ```
+  
+  > [!NOTE]
+  > The framework may support library projects without an entry point in a future release of .NET.
+
+* Enable the <xref:Microsoft.Build.Tasks.Csc.AllowUnsafeBlocks> property, which permits the code generator in the Roslyn compiler to use pointers for JS interop:
+
+  ```xml
+  <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+  ```
+
+  > [!WARNING]
+  > The JS interop API requires enabling <xref:Microsoft.Build.Tasks.Csc.AllowUnsafeBlocks>. Be careful when implementing your own unsafe code in .NET apps, which can introduce security and stability risks. For more information, see [Unsafe code, pointer types, and function pointers](/dotnet/csharp/language-reference/unsafe-code).
+
+  In apps generated from the `wasmbrowser` or `wasmconsole` templates, the <xref:Microsoft.Build.Tasks.Csc.AllowUnsafeBlocks> property is set in the project file (`.csproj`).
+
+* For the .NET 7 release, you must specify `WasmMainJSPath` to point to a file on disk. This file is published with the app, but use of the file isn't required if you're integrating .NET into an existing JS app. We might make this property optional in the future. For more information, see [Smooth out support for running .NET from JS via WebAssembly (dotnet/runtime #77191)](https://github.com/dotnet/runtime/issues/77191).
+
+  In the following example, the JS file on disk is `main.js`, but any JS filename is permissable:
+
+  ```xml
+  <WasmMainJSPath>main.js</WasmMainJSPath>
+  ```
+
+* During preview, set the C# language version:
+
+  ```xml
+  <LangVersion>preview</LangVersion>
+  ```
+
+* Deploy JS files (`.js`) and CSS files (`.css`):
+
+  ```xml
+  <ItemGroup>
+    <WasmExtraFilesToDeploy Include="index.html" />
+    <WasmExtraFilesToDeploy Include="*.js" />
+    <WasmExtraFilesToDeploy Include="*.css" />
+  </ItemGroup>
+  ```
+
```

Comment:
This isn't technically required and the `WasmExtraFilesToDeploy` item group may change in the future, so for now I prefer to leave it undocumented for now.

```suggestion
```